src/Utilities/OperatingSystemFilterHelpers.js

```diff
@@ -1,31 +1,50 @@
+import set  from 'lodash/set';
+import omit  from 'lodash/omit';
+import mapValues  from 'lodash/mapValues';
 import { coerce, compare, rcompare } from 'semver';
-import { OS_CHIP } from './index';
-
-/* Takes an array of string versions and returns an object in the format
-   required by ConditionalFilter component (group filter) */
-export const toGroupSelectionValue = (osVersions = []) => osVersions.reduce((acc, version) => {
-    const [majorVersion, minorVersion] = version.split('.');
-    acc[`${majorVersion}.0`] = {
-        ...(acc[`${majorVersion}.0`] || {}),
-        [`${majorVersion}.${minorVersion}`]: true
-    };
-    return acc;
-}, {});
-
-const compareVersions = (a, b, asc = true) =>
+import { OS_CHIP } from './constants';
+
+export const updateGroupSelectionIdentifier = (selection, major) =>
+    // if every minor version is selected, then mark the group as selected
+    set(selection, [major, major], Object.values({ ...selection[major] }).filter(v => v !== major).every(Boolean));
```

Comment:
This line of code has multiple loops iterating over selection and major. Is there any way to do this in lesser array iterations? But I like that this looks readable.

Reply:
@mkholjuraev, your suggestion does makes sense, however I would vote for keeping this initial version. Optimizing it by reducing cycles is a good thing, but when it comes to arrays with ~10 values at maximum, it doesn't make any difference in performance. Also I find this more readable instead of doing it in one cycle with complex OR/AND chaining and using of `reduce`... I hope I understood you  right